script/set-version

```diff
@@ -1,30 +1,30 @@
-#!/usr/bin/env bash
+#!/bin/sh -eux
 
 set -euo pipefail
 
-pvp="$1"
-semantic="$2"
-
-if [ -z "$pvp" ] || [ -z "$semantic" ]; then
+if [ $# -ne 2 ]; then
     echo "Usage: bump-version <VERSION> <DEPENDENCY>"
     exit 1
 fi
 
+pvp="$1"
+semantic="$2"
+
 specs=(
  amazonka/amazonka.cabal
  core/amazonka-core.cabal
  examples/amazonka-examples.cabal
  gen/amazonka-gen.cabal
  test/amazonka-test.cabal
- amazonka-s3-encryption/amazonka-s3-encryption.cabal
```

Comment:
I had a query about this set of cabal files and why these particular ones are included.
My intuition is that it should include all cabal files. @brendanhay

Reply:
The `set-version` script deals with the hand-rolled `cabal` files, not the generated ones. The flow is to run `set-version` to bump the listed `cabal` file versions, and then run the generator which will discover those versions via:

- [cabal.ede](https://github.com/brendanhay/amazonka/blob/afeb929c758c772fe83692a54978cf7bb05a9de5/gen/template/cabal.ede#L58) - Template for generated `amazonka-*/amazonka-*.cabal` files.
- [Makefile](https://github.com/brendanhay/amazonka/blob/afeb929c758c772fe83692a54978cf7bb05a9de5/gen/Makefile#L9-L15) - which passes the versions from `amazonka` and `amazonka-core` into the generator.

Part of the reason for this (and why `set-version` takes two version numbers) is to allow dependency versions to differ/go out of sync. So far I've only used this a couple of times when pushing out a hotfix to either `amazonka-core` or `amazonka-ec2`, separately.

Reply:
👍 thanks for the explanation.